### DISCUSSION TITLE
RFC3986

### DISCUSSION THREAD

**Discussion by spralja** [collective/icalendar#531] [Q&A]
[RFC5545 ](https://icalendar.org/iCalendar-RFC-5545/3-3-13-uri.html) specifies that the URI value data type must follow the generic URI syntax defined in [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986).  
In [vUri](https://icalendar.readthedocs.io/en/latest/_modules/icalendar/prop.html#vUri) it is stated that:
>Uniform resource identifier is basically just an unquoted string.   

Is this strictly the case?  
If not what would be an example of an invalid URI string?

**Comment by niccokunzmann** [comment]
From my perspective, there seem to be no restrictions on UID.

However, https://www.rfc-editor.org/rfc/rfc7986.html#section-5.3 states:

> This specification updates
   [[RFC5545](https://www.rfc-editor.org/rfc/rfc5545)] by stating that "UID" values MUST NOT include any data that
   might identify a user, host, domain, or any other security- or
   privacy-sensitive information.  It is RECOMMENDED that calendar user
   agents now generate "UID" values that are hex-encoded random
   Universally Unique Identifier (UUID) values as defined in
   Sections [4.4](https://www.rfc-editor.org/rfc/rfc7986.html#section-4.4) and [4.5](https://www.rfc-editor.org/rfc/rfc7986.html#section-4.5) of [[RFC4122](https://www.rfc-editor.org/rfc/rfc4122)].

See also
- https://github.com/collective/icalendar/issues/655
- `icalendar.UIDGenerator`
